lib/workos/user_management.rb

```diff
@@ -10,10 +10,105 @@ module WorkOS
 
   # rubocop:disable Metrics/ModuleLength
   module UserManagement
+    module Types
+      # The ProviderEnum is type-safe declaration of a
+      # fixed set of values for User Management Providers.
+      class Provider < T::Enum
+        enums do
+          Google = new('GoogleOAuth')
+          Microsoft = new('MicrosoftOAuth')
+          AuthKit = new('authkit')
+        end
+      end
+    end
+
     class << self
       extend T::Sig
       include Client
 
+      PROVIDERS = WorkOS::UserManagement::Types::Provider.values.map(&:serialize).freeze
```

Comment:
Now that `Types` is defined in the same namespace, we don't need the full module path:

```suggestion
      PROVIDERS = Types::Provider.values.map(&:serialize).freeze
```